app/apollo/resolvers/cluster.js

```diff
@@ -436,12 +514,21 @@ const clusterResolvers = {
       try {
         logger.info({req_id, user, org_id, cluster_id}, `${queryName} validating`);
 
-        await validAuth(me, org_id, ACTIONS.DETACH, TYPES.CLUSTER, queryName, context);
-
         validateString( 'org_id', org_id );
         validateString( 'cluster_id', cluster_id );
 
-        logger.info({req_id, user, org_id, cluster_id}, `${queryName} saving`);
+        const cluster = await models.Cluster.findOne({
+          org_id,
+          cluster_id
+        }).lean({ virtuals: true });
+
+        await validAuth(me, org_id, ACTIONS.DETACH, TYPES.CLUSTER, queryName, context, [cluster_id, cluster.registration.name || cluster.name]);
```

Comment:
This will throw an exception if `cluster` was not found, due to using `cluster.registration.name`.

Reply:
fix coming